plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/utils/FlexibleEngineMonitorConverter.java

```diff
@@ -83,79 +81,65 @@ private StringBuilder getQueryListMetricsBasicUrl(String region, String projectI
-                    Map<String, String> paramsMap =
-                            getFlexibleEngineMonitorParams(resourceMetricRequest,
-                                    flexibleEngineNameSpaceKind);
-                    for (Map.Entry<String, String> entry : paramsMap.entrySet()) {
-                        url.append(entry.getKey())
-                                .append("=").append(entry.getValue()).append("&");
-                    }
-                    url.deleteCharAt(url.length() - 1);
-                    urlTypeMap.put(url.toString(), monitorType);
-                }
-            } else {
-                StringBuilder url = new StringBuilder(basicUrl);
-                Map<String, String> paramsMap =
-                        getFlexibleEngineMonitorParams(resourceMetricRequest,
-                                flexibleEngineNameSpaceKind);
-                for (Map.Entry<String, String> entry : paramsMap.entrySet()) {
-                    url.append(entry.getKey())
-                            .append("=").append(entry.getValue()).append("&");
-                }
-                url.deleteCharAt(url.length() - 1);
-                urlTypeMap.put(url.toString(), type);
+            StringBuilder url = new StringBuilder(basicUrl);
+            Map<String, String> paramsMap =
+                    getFlexibleEngineMonitorParams(resourceMetricRequest,
+                            metricInfo);
+            for (Map.Entry<String, String> entry : paramsMap.entrySet()) {
+                url.append(entry.getKey())
+                        .append("=").append(entry.getValue()).append("&");
             }
+            url.deleteCharAt(url.length() - 1);
+            return url.toString();
         }
-        return urlTypeMap;
+        return basicUrl;
     }
 
     /**
      * Convert response body to Metric object.
      *
      * @param deployResource The deployed resource.
-     * @param type           MonitorResourceType.
      * @param response       ShowMetricDataResponse.
      * @return Returns Metric object.
      */
     public Metric convertResponseToMetric(DeployResource deployResource,
-                                          MonitorResourceType type,
                                           ShowMetricDataResponse response,
-                                          FlexibleEngineNameSpaceKind flexibleEngineNameSpaceKind) {
+                                          MetricInfoList metricInfo) {
         Metric metric = new Metric();
-        metric.setName(getMetricName(type, flexibleEngineNameSpaceKind));
+        metric.setName(metricInfo.getMetricName());
         Map<String, String> labels = new HashMap<>();
         labels.put("id", deployResource.getResourceId());
         labels.put("name", deployResource.getName());
         metric.setLabels(labels);
         metric.setType(MetricType.GAUGE);
+        if (metricInfo.getUnit().equals("%")) {
+            metric.setUnit(MetricUnit.PERCENTAGE);
```

Comment:
metricUnit is a mandatory field. We must set this value for all cases.

Reply:
Ok.